### ISSUE TITLE
Suggestion - database Locked error

### ISSUE THREAD

**Issue by LazzaAU** [project-alice-assistant/ProjectAlice#302]
when providing input to Alice using a "UserRandomAnswer" intent, depending on what's going on in the background ... telemetry capture, reminder update, hourly update ?? there is a strong possibility of getting database locked error.

Suggest NLUnotRecognised event checks if database is un locked before attempting to write to the DB ? if it is delay it for a few seconds perhaps ?

**Comment by philipp2310** [comment]
DatabaseManager should wait until a timeout is hit before failing.
Worst case: play elevator music until DB is available and continue after 20 seconds?!

**Comment by LazzaAU** [comment]
laughs @ elevator music

**Comment by philipp2310** [comment]
default timeout by sqlite is already 5 seconds - no process should be locking the db that long

**Comment by philipp2310** [comment]
sum of database accesses by HA were over 5 seconds, might cause the locked DB/timeout.

While changing HA skill is a possible start, adjustments for the db are required

**Comment by Psychokiller1888** [comment]
yes but no... DB conn and cursor are closed after after every transaction

**Comment by Psychokiller1888** [comment]
Except for a few misshandeld places, like a fetch that would fail with a `Error fetching data for component` which I'm gonna fix now

**Comment by philipp2310** [comment]
was caused by write access to usb instead of sd and lots of db activities onFiveMinutes in a skill.

- possible enhancement: in memory DB?
